### PR TITLE
Use gpgv for signature verification in dev Dockerfile

### LINKED ISSUE #6526: Use gpgv for signature verification in dev Dockerfile
## Description

https://github.com/freedomofpress/securedrop/blob/develop/securedrop/dockerfiles/focal/python3/Dockerfile

In the dev Dockerfile, we download a few things and then verify their signatures using `gpg --verify`. We should use `gpgv` instead, it's lighter and built to just do one thing, verification. We can also have it verify signatures against just the key we want with `--keyring` instead of having to grep the gpg output for the key that did the verification.

https://github.com/freedomofpress/kernel-builder/commit/782f04a8691773918f58e94a84ccfc52433f1c26 is an example of how I converted from gpg to gpgv in another repository.

### PR DESCRIPTION
**Pull request by noumxn**
update `gpg --verify` to `gpgv --keyring` for verification of signatures

## Status

Work in progress

## Description of Changes
- add `./tor.keyring`, `./firefox.keyring` and `./geckodriver.keyring` to store keys of tor, firefox and geckodriver respectively
- update `gpg --verify` to `gpgv --keyring` for tor, firefox and geckodriver
- delete the keys after verification is complete

Fixes #6526

Changes proposed in this pull request:

## Testing

How should the reviewer test this PR?
Write out any special testing steps here.

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

### If you made changes to the server application code:

- [ ] Linting (`make lint`) and tests (`make test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [ ] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you added or removed a file deployed with the application:

- [ ] I have updated AppArmor rules to include the change

### If you made non-trivial code changes:

- [ ] I have written a test plan and validated it for this PR

Choose one of the following:

- [ ] I have opened a PR in the [docs repo](https://github.com/freedomofpress/securedrop-docs) for these changes, or will do so later
- [ ] I would appreciate help with the documentation
- [x] These changes do not require documentation

Choose one of the following:

- [ ] I have performed a diff review and pasted the contents to [the packaging wiki](https://github.com/freedomofpress/securedrop-debian-packaging/wiki)
- [x] I would like someone else to do the diff review

### CODE REVIEW

**Comment by noumxn** [comment]
Hey, @legoktm. I closed the previous PR due to some issues. I have made some changes to the code, however I am facing difficulty testing them out as I am not working on a linux based os. Could you please help me out with this? Thank you so much.

File: securedrop/dockerfiles/focal/python3/Dockerfile [hunk 1]
```diff
@@ -6,16 +6,17 @@ ARG USER_ID
...
-                       python3-pip python3-all python3-venv virtualenv libpython3.8-dev libssl-dev \
-                       gnupg2 redis-server git xvfb curl wget \
-                       x11vnc enchant libffi-dev sqlite3 gettext sudo \
-                       # For html5validator.  Used only in "app-page-layout-tests", but we can live
-                       # with its being installed along with everything else since it will be
-                       # cached along with everything else too.
-                       default-jdk \
-                       libasound2 libdbus-glib-1-2 libgtk2.0-0 libfontconfig1 libxrender1 \
-                       libcairo-gobject2 libgtk-3-0 libstartup-notification0 tor basez
+                     libgtk2.0 apache2-dev coreutils devscripts vim \
```
Comment (legoktm) [inline securedrop/dockerfiles/focal/python3/Dockerfile]: Can you keep the indentation the same? That will make the patch easier to review.

File: securedrop/dockerfiles/focal/python3/Dockerfile [hunk 2]
```diff
@@ -6,16 +6,17 @@ ARG USER_ID
...
+                     python3-pip python3-all python3-venv virtualenv libpython3.8-dev libssl-dev \
+                     ubuntu-keyring \
```
Comment (legoktm) [inline securedrop/dockerfiles/focal/python3/Dockerfile]: Why did you add the ubuntu-keyring package? I don't think we need it.

We also don't need to add `gpgv`, it's already installed in the base image.

File: securedrop/dockerfiles/focal/python3/Dockerfile [hunk 3]
```diff
@@ -31,9 +32,10 @@ RUN curl -s https://openpgpkey.torproject.org/.well-known/openpgpkey/torproject.
 RUN TBB_VERSION=$(curl -s https://aus1.torproject.org/torbrowser/update_3/release/Linux_x86_64-gcc3/x/en-US | grep -oP 'appVersion="\K[^"]*' | head -1) && \
     wget https://www.torproject.org/dist/torbrowser/${TBB_VERSION}/tor-browser-linux64-${TBB_VERSION}_en-US.tar.xz && \
     wget https://www.torproject.org/dist/torbrowser/${TBB_VERSION}/tor-browser-linux64-${TBB_VERSION}_en-US.tar.xz.asc && \
-    gpg2 --verify tor-browser-linux64-${TBB_VERSION}_en-US.tar.xz.asc 2>&1 | grep "Primary key fingerprint:" | sed -e 's/Primary key fingerprint: //' -e 's/ //g' | tail -1 | grep -qE "${TOR_RELEASE_KEY_FINGERPRINT}" && \
+    gpg2 --output ./tor.keyring --export ${TOR_RELEASE_KEY_FINGERPRINT} && \
+    gpgv --keyring ./tor.keyring ~/Downloads/tor-browser-linux64-${TBB_VERSION}_en-US.tar.xz.asc ~/Downloads/tor-browser-linux64-${TBB_VERSION}_en-US.tar.xz && \
```
Comment (legoktm) [inline securedrop/dockerfiles/focal/python3/Dockerfile]: I'm assuming the `~/Downloads/` is an accidental copy-paste from you testing it? That's why CI is failing.

Also, once we've verified it matches, we should delete `tor.keyring`.

File: securedrop/dockerfiles/focal/python3/Dockerfile [hunk 4]
```diff
@@ -42,14 +44,17 @@ RUN curl -s https://archive.mozilla.org/pub/firefox/releases/${FF_VERSION}/KEY |
...
+    gpgv --keyring ./firefox.keyring ~/Downloads/firefox-${FF_VERSION}.tar.bz2.asc ~/Downloads/firefox-${FF_VERSION}.tar.bz2 && \
     tar xjf firefox-*.tar.bz2 && \
     mv firefox /usr/bin
 
 # Install geckodriver
 RUN wget https://github.com/mozilla/geckodriver/releases/download/${GECKODRIVER_VERSION}/geckodriver-${GECKODRIVER_VERSION}-linux64.tar.gz && \
     wget https://github.com/mozilla/geckodriver/releases/download/${GECKODRIVER_VERSION}/geckodriver-${GECKODRIVER_VERSION}-linux64.tar.gz.asc && \
-    gpg2 --verify geckodriver-${GECKODRIVER_VERSION}-linux64.tar.gz.asc && \
+    # geckodriver uses the same key used by Mozilla
+    gpg2 --keyring ./geckodriver.keyring --export ${MOZILLA_RELEASE_KEY_FINGERPRINT} && \
```
Comment (legoktm) [inline securedrop/dockerfiles/focal/python3/Dockerfile]: Instead of re-exporting a new one for geckodriver, we can just use the same `firefox.keyring`. Maybe name it `mozilla.keyring`?

Big +1 to adding the comment, that's a great idea.

**Review by legoktm — CHANGES_REQUESTED** [review]
Looking good, just a few things need fixing!

**Review by legoktm — APPROVED** [review]
LGTM, will merge when CI passes. Congrats and thank you on your first patch to SecureDrop! 🚀

**Comment by noumxn** [comment]
@legoktm. I dont know if these changes require documentation. I haven't been able to locate where these changes would be documented in the docs repo. Please let me know if any documentation is required and I will get on that ASAP.

**Comment by legoktm** [comment]
I don't think any documentation is needed, this is a fairly internal change.
